core/tee/tee_obj.c

```diff
@@ -53,6 +53,25 @@ TEE_Result tee_obj_get(struct tee_ta_ctx *ctx, uint32_t obj_id,
 	return TEE_ERROR_BAD_PARAMETERS;
 }
 
+TEE_Result tee_obj_get1(struct tee_ta_ctx *ctx, uint32_t obj_id,
+		       struct tee_obj **obj)
+{
+	struct tee_obj *o;
+
+	TAILQ_FOREACH(o, &ctx->objects, link) {
+		if (obj_id == (uint32_t) o) {
+			if (o->info.objectType == TEE_TYPE_CORRUPTED_OBJECT) {
+				tee_obj_close(ctx, o);
```

Comment:
1) What's the problem? That TEE_TYPE_CORRUPTED_OBJECT may disappear in future revisions of the standard? Either we can change something then or we hide it as an internal object type, probably the latter since we don't want to break the ABI.
2) If the handle is closed (either in TEE Core or in the TA) the handle will not be a valid handle any longer (until it's reused, but that must be allowed or the TAs will be be limited in the number of handles they can create and destroy),  using an invalid handle will result in an error code that libutee will panic on.

From what I've seen so far can TEE_GetObjectInfo1() be implemented on top of TEE_GetObjectInfo() in libutee. What matters to the standard is the interface provided.